### PR TITLE
test: add Jest snapshot integration tests 

### PR DESCRIPTION
**Pull request by Krishks369**
Added jest snapshot tests for the template

### CODE REVIEW

File: test/integration.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
+const path = require('path');
+const Generator = require('@asyncapi/generator');
+const {readFile} = require('fs').promises;
+const crypto = require('crypto');
```
Comment (derberg) [inline test/integration.test.js:4]: you can remove it as you do not use it

File: test/integration.test.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+await generator.generateFromString( await asyncapiFile.text()) ;    
+const expectedFiles = [
+    'README.md',
+    'template/src/api/index.js',
+    'template/src/api/routes.js',
+    'template/src/lib/asyncapi.js',
+    'template/src/lib/colors.js',
+    'template/src/lib/config.js',
+    'template/src/lib/path.js'
+
```
Comment (derberg) [inline test/integration.test.js:38]: please remove these white spaces, maybe run it through prettier or something

File: test/integration.test.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
+const path = require('path');
+const Generator = require('@asyncapi/generator');
+const {readFile} = require('fs').promises;
+const crypto = require('crypto');
+const fetch = require('node-fetch');
+const fs = require('fs');
+
+const MAIN_TEST_RESULT_PATH = path.join('test', 'temp',' integrationTestResult');
+const URL = 'https://raw.githubusercontent.com/Krishks369/shrekapp-asyncapi-designed/main/asyncapi.yaml'
```
Comment (derberg) [inline test/integration.test.js:9]: maybe we should use some official example, like https://raw.githubusercontent.com/asyncapi/spec/master/examples/social-media/frontend/asyncapi.yaml or https://github.com/asyncapi/generator/blob/master/test/docs/ws.yml ?

File: package.json [hunk 4]
```diff
@@ -16,32 +16,41 @@
...
   "publishConfig": {
     "access": "public"
   },
   "dependencies": {
     "@asyncapi/generator-filters": "^2.1.0",
-    "@asyncapi/generator-hooks": "^0.1.0"
+    "@asyncapi/generator-hooks": "^0.1.0",
+    "@asyncapi/parser": "^1.13.1",
+    "dummy-json": "^3.0.4",
+    "node-yaml-config": "^1.0.0"
```
Comment (derberg) [inline package.json:41]: I don't think you are using these

File: package.json [hunk 5]
```diff
@@ -16,32 +16,41 @@
     "type": "git",
     "url": "git+https://github.com/asyncapi/nodejs-ws-template.git"
   },
+  "type": "module",
```
Comment (derberg) [inline package.json:19]: just curious, what does it change?

File: package.json [hunk 6]
```diff
@@ -16,32 +16,41 @@
...
   "homepage": "https://github.com/asyncapi/nodejs-ws-template#readme",
   "scripts": {
     "release": "semantic-release",
     "lint": "echo 'No linter configured yet'",
-    "test": "echo 'No tests yet'",
     "generate:assets": "npm run generate:readme:toc",
     "generate:readme:toc": "markdown-toc -i README.md",
-    "bump:version": "npm --no-git-tag-version --allow-same-version version $VERSION"
+    "bump:version": "npm --no-git-tag-version --allow-same-version version $VERSION",
+    "start": "node src/api/index.js",
```
Comment (derberg) [inline package.json:30]: what does it start?

**Review by derberg — CHANGES_REQUESTED** [review]
Looks good mate, added just few comments, some stuff are not needed

also remember to later commit updated package-lock, without removed deps

**Comment by Krishks369** [comment]
> Looks good mate, added just few comments, some stuff are not needed
> 
> also remember to later commit updated package-lock, without removed deps
 
I don't understand what you meant by deps?

**Comment by derberg** [comment]
@Krishks369 oh sorry, by `deps` I meant `dependencies` 😅

**Comment by Krishks369** [comment]
@derberg Made the changes you asked. Check it now

File: test/integration.test.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+        const params = {
+			server: 'localhost'
+		}
+        const generator = new Generator(path.normalize('./'), outputDir, {
+			forceWrite: true,
+			templateParams: params
+		});
+		await generator.generateFromString(await asyncapiFile.text());
+		const expectedFiles = [
+			'README.md',
```
Comment (derberg) [inline test/integration.test.js:29]: I just noticed that you are testing the wrong files. You test template sources, while you should test the output created from these templates.

snapshot integration tests purpose is to show you how the output changes basing on your changes to the template files.

so you should check `src/api/index.js` and others, but not under `template` but under the location where all gets generated, `outputDir`

Comment (Krishks369) [reply test/integration.test.js:29]: Oh yea I will rectify it. Also do you want me to test docker and readme file also?

Comment (derberg) [reply test/integration.test.js:29]: no, enough is the service related files, js code

**Comment by derberg** [comment]
I updated the PR title a bit to follow imperative mood  -> https://cbea.ms/git-commit/#imperative

**Comment by derberg** [comment]
Interesting, I see this in test failure:
```
- router.ws('C:\echo', async (ws, req) => {
+ router.ws('/echo', async (ws, req) => {
```

I think it is related to https://github.com/asyncapi/nodejs-ws-template/issues/103
Can you try the workarounds mentioned there?

File: filters/all.js [hunk 8]
```diff
@@ -10,7 +10,12 @@ function port(url, defaultPort) {
 filter.port = port;
 
 function pathResolve(pathName, basePath = '/') {
-  return path.resolve(basePath, pathName);
+  if(pathName.startsWith("/")){
+  return pathName;
```
Comment (derberg) [inline filters/all.js:14]: oh this indent hurts my eyes 😄 ? sorry 😅
anyway, fault of the project that has no eslint not yours, 👍🏼 from my side

 if you want to contribute further to this project, plugging in eslint would be a good contribution, just take config and plugins that we use here https://github.com/asyncapi/generator

**Review by derberg — APPROVED** [review]
thanks a lot for this contribution! super useful!

**Comment by derberg** [comment]
@all-contributors please add @Krishks369 for test

**Pull request merged**